File: 1277-count-square-submatrices-with-all-ones/1277-count-square-submatrices-with-all-ones.java
class Solution {
    public int countSquares(int[][] matrix) {
        int m = matrix.length, n = matrix[0].length;
        int total = 0;
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                if(matrix[i][j] == 0) continue;
                int l = (j - 1 < 0) ? 0 : matrix[i][j-1];
                int t = (i - 1 < 0 )? 0 : matrix[i-1][j];
                int d = (i - 1 < 0 || j - 1 < 0 ) ? 0 : matrix[i-1][j-1];
                
                int min = Math.min(l,t); min = Math.min(min,d);
                matrix[i][j] = min + 1;
                total += matrix[i][j];
                
            }
        }
        
        return total;
    }
}

File: 1277-count-square-submatrices-with-all-ones/1277-count-square-submatrices-with-all-ones.py
class Solution:
    def countSquares(self, matrix: List[List[int]]) -> int:
        m,n = len(matrix), len(matrix[0])
        total = 0
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 1:
                    adj_min = min(matrix[i-1][j-1],matrix[i-1][j], matrix[i][j-1]) if i -1 >= 0 and j - 1 >= 0 else 0
                    matrix[i][j] = adj_min + 1
        for i in range(m):
            for j in range(n):
                    total += matrix[i][j]
        #print(matrix)
        return total
    
    
"""

[0, 1, 1, 1]
[1, 1, 2, 2]
[0, 1, 2, 3]]


"""